src/commonMain/kotlin/fr/acinq/lightning/serialization/v4/Deserialization.kt

```diff
@@ -254,6 +254,7 @@ object Deserialization {
             pubkeyScript = readDelimitedByteArray().toByteVector(),
             localAmount = readNumber().msat,
             remoteAmount = readNumber().msat,
+            htlcAmount = readNumber().msat
```

Comment:
That's not backwards-compatible, is it? You must introduce a new discriminator (e.g. `0x02`) for which we encode the `htlcAmount`. When you encounter the `0x01` discriminator, you should just fill this with `0.msat`.

Reply:
Fixed in [99ff0f6](https://github.com/ACINQ/lightning-kmp/pull/568/commits/99ff0f693fc569bb89008a97791f5340d6b8478d). But it makes me nervous that we don't have any serialization tests that would have caught this.

Reply:
You should definitely add one! :)

Reply:
Added in [b72530d](https://github.com/ACINQ/lightning-kmp/pull/568/commits/b72530df3c547ac67a20058afbba959aa6b8835e).